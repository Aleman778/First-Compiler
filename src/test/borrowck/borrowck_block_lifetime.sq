// ignore

fn main() {
    let a: &i32;
    {
        let b: i32 = 5;
        a = &b;
    }
    print_int(*a);
}
