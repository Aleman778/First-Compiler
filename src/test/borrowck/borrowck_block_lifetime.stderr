error: `b` does not live long enough
  --> $DIR/borrowck_block_lifetime.sq:7:13
   |
LL |         a = &b;
   |             ^^ borrowed value does not live long enough
LL |     }
   |     - `b` dropped here while still borrowed
LL |     print_int(*a);
   |               -- borrow later used here